seqr/views/apis/anvil_workspace_api.py

```diff
@@ -193,29 +188,94 @@ def create_project_from_workspace(request, namespace, name):
 
     project = create_model_from_json(Project, project_args, user=request.user)
 
+    _trigger_add_workspace_data(project, pedigree_records, request.user, request_json['fullDataPath'], request_json['sampleType'])
+
+    return create_json_response({'projectGuid': project.guid})
+
+
+@anvil_auth_and_policies_required
+def add_workspace_data(request, project_guid):
+    """
+    Add data from an AnVIL workspace.
+
+    :param request: Django request object
+    :param project_guid: Django request object
+    :return success if no exceptions
+
+    """
+    project = get_project_and_check_permissions(project_guid, request.user, can_edit=True)
+    check_workspace_perm(request.user, CAN_EDIT, project.workspace_namespace, project.workspace_name, can_share=True)
```

Comment:
this is redundant, `get_project_and_check_permissions` will call `has_workspace_perm` if there isn't other access, as will `check_workspace_perm` so there is no case where `check_workspace_perm` is True but `get_project_and_check_permissions` fails. It would be better to just get the project directly without checking permission and the call `check_workspace_perm`

Reply:
I see what you mean. Change to get the project directly.